omc3/segment_by_segment/propagables/abstract.py

```diff
@@ -0,0 +1,152 @@
+
+"""
+Abstract Propagable
+-------------------
+In this module the abstract class for a propagable is defined.
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from functools import cache
+from typing import TYPE_CHECKING
+
+import pandas as pd
+
+from omc3.definitions.constants import PLANES
+from omc3.definitions.optics import OpticsMeasurement
+from omc3.segment_by_segment import propagables  # don't import alpha/beta etc directly ! cyclic imports !
```

Comment:
Oof, big smell that we are potentially doing cyclic imports...

Reply:
As this problem only comes here in the Abstract class and I should not need to touch this code again, I think we will be fine. Also you will notice immediately with the tests